administrator/components/com_media/Controller/Api.php

```diff
@@ -104,7 +105,20 @@ public function files()
 			switch ($method)
 			{
 				case 'get':
-					$data = $this->getModel()->getFiles($adapter, $path, $this->input->getWord('filter'));
+					$option = $this->input->getWord('opt');
+
+					switch ($option)
+					{
+						case 'permalink':
+							$data = $this->getModel()->getUrl($adapter, $path);
+							break;
+
+						default:
+							$data = $this->getModel()->getFiles($adapter, $path, $this->input->getWord('filter'));
+							break;
+
+					}
+
```

Comment:
@kasvith I think an if else is better here (`if (isset($option) && $option === 'permalink')`